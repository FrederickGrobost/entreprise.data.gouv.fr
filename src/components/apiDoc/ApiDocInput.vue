<template>
  <div class="apidoc__input">
    <input v-model="requestToDisplay" @keyup.enter="activateButton" />
    <a ref="goToButton" class="button" :href="requestToDisplay">GET</a>
  </div>
</template>

<script>
export default {
  name: "ApiDocInput",
  props: {
    request: { type: String, default: "https://entreprise.data.gouv.fr/" }
  },
  data() {
    return {
      requestToDisplay: this.request
    };
  },
  methods: {
    // Pressing enter will activate the click
    activateButton() {
      this.$refs.goToButton.click();
    }
  }
};
</script>

<style lang="scss" scoped>
.apidoc__input {
  display: flex;
  flex-direction: row;
}
</style>
