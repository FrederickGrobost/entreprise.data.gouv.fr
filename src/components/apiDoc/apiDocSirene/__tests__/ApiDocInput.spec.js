import { createLocalVue, mount, RouterLinkStub } from "@vue/test-utils";
import { __createMocks as createStoreMocks } from "@/store/index.js";
import Vuex from "vuex";
import ApiDocInput from "@/components/apiDoc/ApiDocInput";

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.component("router-link", RouterLinkStub);
jest.mock("@/store/index.js");

describe("ApiDocInput.vue", () => {
  const storeMocks = createStoreMocks();
  const wrapperADI = mount(ApiDocInput, {
    localVue,
    store: storeMocks.store
  });

  test("Method activateButton activates the button", () => {
    wrapperADI.vm.$refs.goToButton.click = jest.fn();
    wrapperADI.vm.activateButton();
    expect(wrapperADI.vm.$refs.goToButton.click).toHaveBeenCalled();
  });
});

describe("ApiDocInput.vue snapshot testing", () => {
  const storeMocks = createStoreMocks();
  const wrapperADI = mount(ApiDocInput, {
    localVue,
    store: storeMocks.store
  });

  test("It match the snapshot", () => {
    expect(wrapperADI.vm.$el).toMatchSnapshot();
  });
});
