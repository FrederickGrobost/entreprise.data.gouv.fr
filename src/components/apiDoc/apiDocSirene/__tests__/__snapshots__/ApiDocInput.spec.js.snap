// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiDocInput.vue snapshot testing It match the snapshot 1`] = `
<div
  class="apidoc__input"
>
  <input />
   
  <a
    class="button"
    href="https://entreprise.data.gouv.fr/"
  >
    GET
  </a>
</div>
`;
